hibernate-core/src/main/java/org/hibernate/cfg/EJB3DTDEntityResolver.java

```diff
@@ -134,10 +155,46 @@ private InputSource buildInputSource(String publicId, String systemId, InputStre
 		return source;
 	}
 
-	private InputStream getStreamFromClasspath(String fileName) {
-		LOG.trace( "Recognized JPA ORM namespace; attempting to resolve on classpath under org/hibernate/jpa" );
+	private InputStream getStreamFromJpaClasspath(String fileName) {
+		LOG.tracev( "Recognized JPA ORM namespace; attempting to resolve [{0}] on classpath under org/hibernate/jpa", fileName );
 		String path = "org/hibernate/jpa/" + fileName;
-		InputStream dtdStream = resolveInHibernateNamespace( path );
+		return ResourcesHelper.locateResourceAsStream( path, this.getClass().getClassLoader() );
+	}
+
+	private InputStream getStreamFromNonJpaClasspath(String fileName) {
+		LOG.tracev( "Attempting to resolve [{0}] on classpath under org/hibernate/", fileName );
+		String path = "org/hibernate/" + fileName;
+		InputStream dtdStream = ResourcesHelper.locateResourceAsStream( path, this.getClass().getClassLoader() );
+		if ( dtdStream == null ) {
+			if ( fileName.indexOf( "2.0" ) > -1 ) {
+				LOG.usingOldDtd();
+			}
+		}
 		return dtdStream;
 	}
+
+	// Copied over from ConfigHelper that was removed
+	private InputStream resolveInLocalNamespace(String path) {
+		boolean hasLeadingSlash = path.startsWith( "/" );
+		String stripped = hasLeadingSlash ? path.substring( 1 ) : path;
```

Comment:
I wonder how often there is a difference between look-up with the leading slash and without.

Anyway, the `ResourceHelper` methods already handle this.  So why not just call `ResourceHelper` here as well?

Also, why are `getStreamFromJpaClasspath` and `getStreamFromNonJpaClasspath` so different?  They both (should)  effectively do the same thing.

Reply:
> I wonder how often there is a difference between look-up with the leading slash and without.
> 
> Anyway, the `ResourceHelper` methods already handle this. So why not just call `ResourceHelper` here as well?
> 
> Also, why are `getStreamFromJpaClasspath` and `getStreamFromNonJpaClasspath` so different? They both (should) effectively do the same thing.


So, to recap where this comes from, the point was to remove the deprectated DTDEntityResolver, but since the EJB3DTDEntityResolver wasn't deprecated, functionality from the former that the latter depends on had to be moved over. In the same go, deprecate EJB3DTDEntityResolver so that we can remove it in the near future as well.
My primary concern in this refactoring is to do it with minimal risk of introducing new bugs (there is also no test for either class to be able to double check), so i just copied over whatever was needed with that in mind.
That's the motivation of trying to change this code as little as possible and why there still is some difference between 'getStreamFromJpaClasspath' and 'getStreamFromNonJpaClasspath'. I considered the possibility of taking it further, but I don't know the internals of the old code enough to know why this or that is done the way it is so again, trying to respect what there was as much as possible.
I could of course write a test for this class, but I wonder if it's worth the effort, since this is most likely not being used, and now also deprecated.
That said, I'll take the refactoring as far as you want me to, it's more a matter of me not knowing what the general policy is in such cases.